### ISSUE TITLE
Special characters in typewriter

### ISSUE THREAD

**Issue by ElbertCode** [flxzt/rnote#599]
**Is your feature request related to a problem? Please describe.**
The process of inserting special characters, such as accents or mathematical symbols, into text boxes without corresponding keyboard keys is quite cumbersome, as users must first copy them from an external app and then manually insert them into Rnote.

**Describe the solution you'd like**
As there is already a button for inserting emojis, this could be used and extended to include special characters.

**Additional context**
In terms of design/layout, it could be implemented in a similar way to how symbols/emojis are inserted on Windows:

![](https://support.content.office.net/en-us/media/62944d33-68f7-43d2-9ffe-667924ca62af.png)

**Comment by Kneemund** [comment]
> As there is already a button for inserting emojis, this could be used and extended to include special characters.

The emoji picker is a plain GTK4 widget, specifically `GtkEmojiChooser`. This most likely can't be easily extended and would have to be implemented from scratch instead.

**Comment by flxzt** [comment]
I do like the idea, but needs more discussion what should actually be in there. Probably the most common math & greek symbols. What else?

> This most likely can't be easily extended and would have to be implemented from scratch instead.

We could reuse the icon picker though :)

**Comment by ElbertCode** [comment]
> Probably the most common math & greek symbols. What else?

I would also suggest:
- currency signs
- variations of the Latin alphabet (accents, etc.) e.g. from French, Spanish, ...
- perhaps arrow symbols

**Comment by LeSnake04** [comment]
maybe it makes sense to just make a separate button with a custom menu then.

**Comment by PhilDevProg** [comment]
> >> As there is already a button for inserting emojis, this could be used and extended to include special characters.
> >
> > The emoji picker is a plain GTK4 widget, specifically GtkEmojiChooser . This most likely can't be easily extended and would have to be implemented from scratch instead.
> 
>  maybe it makes sense to just make a separate button with a custom menu then.    

There's already RnIconPicker which is used for choosing the icon of the workspaces (I guess; I didn't check) so it could maybe be reused for this?

**Comment by Kneemund** [comment]
Yes, that's what flxzt meant by "forking the icon picker". We should probably use the grouped icon picker though.
